server/db/initDB.ts

```diff
@@ -7,7 +7,7 @@ require('dotenv').config();
 
 export async function initDB() {
 	try {
-		await mongoose.connect(process.env.URI);
+		await mongoose.connect(process.env.URI as string);
```

Comment:
Perhaps, it worth checking the type at first (since it may be not set) and throwing the clear error.
This will help you to debug if someone forgets to set the URI (for 100%) :)